container/src/LuigiContainer.svelte

```diff
@@ -94,12 +102,32 @@
     return !!webcomponent;
   }
 
+  function canMfeInitialized(): boolean {
+    return !!can_mfe_initialized;
+  }
+
   onMount(async () => {
     const ctx = context ? JSON.parse(context) : {};
     if (isWebComponent()) {
       mainComponent.innerHTML = '';
       webcomponentService.renderWebComponent(viewurl, mainComponent, ctx, {});
     }
+    if (canMfeInitialized()) {
+      mfe_initialized_ready = true;
+      setTimeout(() => {
+        dispatchLuigiEvent(Events.INITIALIZED, {});
+      });
+    } else if (isWebComponent()) {
+      mainComponent.addEventListener('wc_ready', () => {
+        if (
+          !(mainComponent as any)._luigi_mfe_webcomponent
+            ?.deferLuigiClientWCInit
+        ) {
+          mfe_initialized_ready = true;
+          dispatchLuigiEvent(Events.INITIALIZED, {});
+        }
+      });
+    }
```

Comment:
The whole thing seems a bit duplicated. https://github.com/SAP/luigi/issues/3344 seems to suggest that this feature should be about WC only, since it seems we handled it for iframes already. So I guess the whole thing should fit under **isWebComponent()** ?:
```
// here
if (isWebComponent) {
      mainComponent.innerHTML = '';
      webcomponentService.renderWebComponent(viewurl, mainComponent, ctx, {});
      mainComponent.addEventListener('wc_ready', () => {
          if (
            !(mainComponent as any)._luigi_mfe_webcomponent
              ?.deferLuigiClientWCInit
          ) {
            mfe_initialized_ready = true;
            dispatchLuigiEvent(Events.INITIALIZED, {});
          }
        });

        if (canMfeInitialized()) {
              mfe_initialized_ready = true;
              setTimeout(() => {
                    dispatchLuigiEvent(Events.INITIALIZED, {});
              });
        }
}
```

Reply:
Talked with Philipp about the implementation. It's fine.
Regarding "duplication":
`if(!(mainComponent as any)._luigi_mfe_webcomponent?.deferLuigiClientWCInit)` checks if webcomponents, which are extending LuigiElement, does not have the option to defer the LuigiClient initialization. In the suggested code the dispatchLuigiEvent would be called twice if a LuigiContainer tag has the property `can_mfe_initialized="true"` and `webcomponent="true` and the webcomponent has no option `deferLuigiClientWCInit` or it is set to false (which is the default).